### ISSUE TITLE
404.html page has broken paths to css and js

### ISSUE THREAD

**Issue by VaclavElias** [stride3d/stride-docs#108]
As I am preparing for the new docs update I discovered issue with the current **404.html** page which has broken **css** and **js** paths. 

The problem was caused by the recent docs update where we decoupled external domain which was hosting css and js files. The styling works across all docs pages, but it broke 404.html page which is located in each version and each language folder (8 files).

The problem is that 404.html needs ideally an absolute path for styling and js but we can't have absolute paths for the remaining pages because of the versioning and languages feature and the individual generated docs are not aware of any language and version. Language and version functionality is happening through the redirects. Also, each version might use different styling and layout.

Ideally, the common assets should be centralised to avoid any future problems. 

**Current Fix**

The fix was applied directly to  **web.config** in the web server by adding 2 rules:

For example it redirects a path `https://doc.stride3d.net/4.1/en/ReleaseNotes/styles/docfx.css` generated from the relative path `styles/docfx.css` to `https://doc.stride3d.net/4.1/en/styles/docfx.css`.

```
        <!-- Fix Deep Asset Paths for 'latest' -->
        <rule name="Fix Deep Asset Paths for latest" stopProcessing="true">
            <match url="^latest/(en|jp)/.+/(styles|css|scripts)/(.*)" />
            <conditions>
                <add input="{REQUEST_FILENAME}" matchType="IsFile" negate="true" />
            </conditions>
            <action type="Rewrite" url="4.1/{R:1}/{R:2}/{R:3}" />
        </rule>

        <!-- Fix Deep Asset Paths for versions -->
        <rule name="Fix Deep Asset Paths for versions" stopProcessing="true">
            <match url="^(\d+\.\d+)/(en|jp)/.+/(styles|css|scripts)/(.*)" />
            <conditions>
                <add input="{REQUEST_FILENAME}" matchType="IsFile" negate="true" />
            </conditions>
            <action type="Rewrite" url="{R:1}/{R:2}/{R:3}/{R:4}" />
        </rule>
```

**Another solution**

I would suggest to centralise **404.html** page to one page, which will be in the root folder, available for all languages and versions, and it would have own styling embedded. 

There are several benefits for this:

1. Simplicity: You only have one 404 page to maintain and update.
1. Consistency: All users, regardless of the documentation version or language they are browsing, will see the same error page, leading to a consistent user experience.
1. Efficiency: Reduces the overhead of managing styles for error pages across multiple directories.
1. Flexibility: In the future, if you need to change the design or message of the 404 page, you only have to do it in one place.

Ideally, this solution should be implemented before the new docs release.

**ToDo**

- [ ]  Create PR to update web.config https://github.com/stride3d/stride-docs/blob/master/build/web.config, we don't need to deploy this file as the file was updated manually on the server already
- [x] Check with xen2 if the centralised 404.html could be implemented

**Comment by VaclavElias** [comment]
Just additional update on this matter..

I came up with another solution which I discussed with xen2. 

We don't need to centralise **404.html** page but I would need to update the power shell script with post processing and update 404.html assets from relative path to absolute path. 404 page is the only page which must have absolute path to their assets. See the image below. What does it mean?

1) I have to update all existing 404.html pages (8 pages) and change the path to e.g. /4.1/en/ or /4.1/jp/

2) Reverse temporary web.config update

3) Update the Stride Docs ps1 which will generate this path for the new 404.html pages. This means each incremental Stride Docs release, would need to have the version updated. It should be ok as this version number will be inside the ps1 file. So if there is a new release e.g. 4.2 you need to first update this parameter before the docs generation and deployment

4) The new Stride Docs would also benefit from this parameter because, we might be able to publish automatically to the 4.2 folder and update the **versions.json**, I might be able to do it through GitHub Actions.

**Comment by VaclavElias** [comment]
Just an update regarding the old docs. I have created PR which fixes 404.html in post processing. This PR might not be needed if I am able to finish new docs, but just in case.. that PR is important if we have to deploy the current docs.

I have also manually updated deployed docs through FTP and corrected specific paths in **404.html** pages (4 files) so the 404 looks ok. There are some errors in the console, which were there also in the past buy they are not causing any issues.
